script.js

```diff
@@ -311,161 +322,200 @@ function paintScreen() {
+
+
 }
-// function pickUpBins(){ //keyP
 
-// }
-//document.addEventListener('keypress');
+
 
 var viewportWidth = window.innerWidth;
 var viewportHeight = window.innerHeight;
 var playKeys = {}; // will store  key-true/false pairs 
 playKeys["Space"] = false;
+playKeys["KeyP"] = false;
 
 function keyBeingPressed(event) {
-    if (event.code != "Space") {
+    if (event.code != "Space" && event.code != "KeyP") {
         playKeys[event.code] = true;
         
     }
 }
 
 
 function keyNotBeingPressed(event) {
-    if (event.code != "Space") {
+    if (event.code != "Space" && event.code != "KeyP") {
         playKeys[event.code] = false;
     }
 }
 
-// function keyWasPressed(event){
-//     playKeys[event.code] = !playKeys[event.code];
-//     console.log(playKeys[event.code]);
-// }
+
 
 document.addEventListener('keydown', keyBeingPressed);
 document.addEventListener('keyup', keyNotBeingPressed);
 
 function movePlayer() { //edited in pickUpStuff
+    var changeInX = 0;
+    var changeInY = 0;
     if (playKeys['KeyW'] && playerY - 5 > 0) {
-        playerY -= 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.y -= 10;
-        }
+        changeInY -=10;
     }
     if (playKeys['KeyS'] && playerY + player.offsetHeight + 5 < viewportHeight) {
-        playerY += 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.y += 10;
-        }
+        changeInY +=10;
     }
 
     if (playKeys['KeyA'] && playerX - 5 > 0) {
-        playerX -= 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.x -= 10;
-        }
+        changeInX -=10;
     }
     if (playKeys['KeyD'] && playerX + player.offsetWidth + 5 < viewportWidth) {
-        playerX += 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.x += 10;
-        }
+        changeInX +=10;
     }
+    playerY += changeInY;
+    playerX += changeInX;
     for (let i in items) {
-        if (items[i] == playerIsCarrying) {
+        if (items[i] == playerIsCarrying && playerIsCarrying != null) {
+            playerIsCarrying.x += changeInX;
+            playerIsCarrying.y += changeInY;
             items[i].x = playerIsCarrying.x;
             items[i].y = playerIsCarrying.y;
             items[i].element.style.left = items[i].x + 'px';
-            items[i].element.style.top = items[i].y; + 'px';
+            items[i].element.style.top = items[i].y + 'px';
         }
     }
+    for (let c in containers){
+        if (containers[c] == playerIsCarrying && playerIsCarrying != null) {
+            playerIsCarrying.x += changeInX;
+            playerIsCarrying.y += changeInY;
+            containers[c].x = playerIsCarrying.x;
+            containers[c].y = playerIsCarrying.y;
+            containers[c].element.style.left = containers[c].x + 'px';
+            containers[c].element.style.top = containers[c].y + 'px';
```

Comment:
I think you'll want to handle this sort of situation like we changed the items to do - this function (`movePlayer`) should do just one of the following:
- update the "game state" (`x` and `y` of the player/items/container - NOT the HTML element)
- or update the "rendered game" (`element.style.left = '3px'`, for example)

You can decide which makes more sense to you, but I think the first option makes more sense - this function just updates the game state, and then the paint functions you've written should handle moving things on screen. Lemme know if you want to talk about this more offline - I do think it's the same thing as what we did to the item moving last week, but it is happening in more places.